scripts/common/proxy.sh

```diff
@@ -1,53 +1,94 @@
-
-function proxy_bootstrap() {
+# read_proxy_config_from_env makes sure that both proxy variables (upper
+# and lower case) are set to the same value. for example http_proxy and
+# HTTP_PROXY both must point to the same address. this function sets the
+# following variables ENV_HTTP_PROXY_ADDRESS, ENV_HTTPS_PROXY_ADDRESS,
+# and ENV_NO_PROXY.
+function read_proxy_config_from_env() {
+    # by default https proxy configuration inherits from http proxy config.
+    # we also make sure that we have both environment variables (with upper
+    # and lower case) set to the same value.
     if [ -n "$HTTP_PROXY" ]; then
-        ENV_PROXY_ADDRESS="$HTTP_PROXY"
-        export https_proxy="$HTTP_PROXY"
```

Comment:
this used to always export https_proxy and now that has changed. i imagine because modern linux tools reliy on that env var and not the others.

Reply:
Are you saying that we should always export HTTPS_PROXY to the HTTP_PROXY value when the first is not set ?

Reply:
i just dont want to change the existing behavior. im worried we will introduce a regression. how about exporting all 4 at the end?

```bash
    if [ -n "$ENV_HTTP_PROXY_ADDRESS" ]; then
        export http_proxy="$ENV_HTTP_PROXY_ADDRESS"
        export HTTP_PROXY="$ENV_HTTP_PROXY_ADDRESS"
    fi
    if [ -n "$ENV_HTTPS_PROXY_ADDRESS" ]; then
        export https_proxy="$ENV_HTTPS_PROXY_ADDRESS"
        export HTTPS_PROXY="$ENV_HTTPS_PROXY_ADDRESS"
    fi
```

Reply:
Sounds good to me. I have added this to the last commit. Haven't tested it yet but I will come back here as soon as I do so.

Reply:
I have tested an install the proxies are being set correctly after the change.